src/IO/S3/BlobStorageLogWriter.cpp

```diff
@@ -54,15 +57,22 @@ void BlobStorageLogWriter::addEvent(
 BlobStorageLogWriterPtr BlobStorageLogWriter::create(const String & disk_name)
 {
 #ifndef CLICKHOUSE_KEEPER_STANDALONE_BUILD /// Keeper standalone build doesn't have a context
-    if (auto blob_storage_log = Context::getGlobalContextInstance()->getBlobStorageLog())
+    const auto & global_context = Context::getGlobalContextInstance();
+    bool enable_blob_storage_log = global_context->getSettingsRef().enable_blob_storage_log;
+    if (auto blob_storage_log = global_context->getBlobStorageLog())
     {
         auto log_writer = std::make_shared<BlobStorageLogWriter>(std::move(blob_storage_log));
 
         log_writer->disk_name = disk_name;
-        if (CurrentThread::isInitialized() && CurrentThread::get().getQueryContext())
+        const auto & query_context = CurrentThread::isInitialized() ? CurrentThread::get().getQueryContext() : nullptr;
+        if (query_context)
+        {
             log_writer->query_id = CurrentThread::getQueryId();
+            enable_blob_storage_log = query_context->getSettingsRef().enable_blob_storage_log;
```

Comment:
May be check `enable_blob_storage_log` from query context before getting `blob_storage_log` and creating `BlobStorageLogWriter`?

Reply:
Move the check to `Context::getBlobStorageLog()`